### ISSUE TITLE
Output of Yolo-Keras

### ISSUE THREAD

**Issue by bhanu223** [experiencor/basic-yolo-keras#46]
1425/3080 [============>.................] - ETA: 1803s - loss: 0.15872017-11-18 10:13:30.536160: I tensorflow/core/kernels/logging_ops.cc:79] Dummy Line    [0]
2017-11-18 10:13:30.536237: I tensorflow/core/kernels/logging_ops.cc:79] Loss XY        [0.0229914095]
2017-11-18 10:13:30.536257: I tensorflow/core/kernels/logging_ops.cc:79] Loss WH        [0.0974636301]
2017-11-18 10:13:30.536286: I tensorflow/core/kernels/logging_ops.cc:79] Loss Conf      [0.00172986195]
2017-11-18 10:13:30.536298: I tensorflow/core/kernels/logging_ops.cc:79] Loss Class     [0.0058139083]
2017-11-18 10:13:30.536310: I tensorflow/core/kernels/logging_ops.cc:79] Total Loss     [0.127998814]
2017-11-18 10:13:30.536321: I tensorflow/core/kernels/logging_ops.cc:79] Current Recall         [0.607142866]
2017-11-18 10:13:30.536334: I tensorflow/core/kernels/logging_ops.cc:79] Average Recall         [0.567189932]
l_bound:1232
r_bound:1248
r_bound:1248
l_bound:1232
1426/3080 [============>.................] - ETA: 1802s - loss: 0.15872017-11-18 10:13:31.626686: I tensorflow/core/kernels/logging_ops.cc:79] Dummy Line    [0]
2017-11-18 10:13:31.626748: I tensorflow/core/kernels/logging_ops.cc:79] Loss XY        [0.0277243108]
2017-11-18 10:13:31.626766: I tensorflow/core/kernels/logging_ops.cc:79] Loss WH        [0.126982644]
2017-11-18 10:13:31.626779: I tensorflow/core/kernels/logging_ops.cc:79] Loss Conf      [0.000806401658]
2017-11-18 10:13:31.626790: I tensorflow/core/kernels/logging_ops.cc:79] Loss Class     [0.0110849598]
2017-11-18 10:13:31.626802: I tensorflow/core/kernels/logging_ops.cc:79] Total Loss     [0.16659832]
2017-11-18 10:13:31.626812: I tensorflow/core/kernels/logging_ops.cc:79] Current Recall         [0.763157904]
2017-11-18 10:13:31.626824: I tensorflow/core/kernels/logging_ops.cc:79] Average Recall         [0.567207873]

Is this correct output or am I missing something. Please help

**Comment by bhanu223** [comment]
Where are the checkpoints getting saved?

**Comment by experiencor** [comment]
@bhanu223 You got this far, then the "Error when checking target: expected lambda_2 to have shape" error's gone?

The trained weight is saved to a file in the root folder? The name of the file can be specified in field "saved_weights_name" in the config.json file. If this field is empty, it's saved to "best_weights.h5". The line to save the weight is located at line 420, frontend.py.

**Comment by bhanu223** [comment]
@experiencor Yeah! Luckily by uninstalling the latest keras and Reinstalling 2.0.8 version worked for me.

I figured out where my chkpt file is!

I have a doubt, I'm detecting vehicle and my training data is VOC dataset(5000 images) where only vehicles are annotated and 1000 images of my own. 
By using Full Yolo weights with 6-7 epochs I'm getting loss less than 0.2 and the model is not converging further and It stopped (due to the "Early" stopping call back). What is the ideal thing to do now?

**Comment by experiencor** [comment]
I suggest you train in two steps:

1. Set warmup_batches to about three times the size of the training set. Wait for the training to end.

2. Set warmup_batches to 0 and train again.

If it's still not working, you may play the 4 scales to increase recall (e.g. increase object_scale and coord_scale).

**Comment by bhanu223** [comment]
Actually I reduced the learning rate by 10times. Now it converged in 15epochs. Batch size is increased to 24. I've got total training loss of 0.06 finally. I have to test the performance.
